misc/server/src/behaviour.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(deprecated)]
+
```

Comment:
What are we deprecating here that can't be fixed in this PR? As a general rule, we should fix warnings in our own code. If we can't then our users won't be able to either.

Reply:
I think that's some relic from the AutoNATv2 changes

Reply:
I removed that